## Conexion a la API de IBM Watson

In [129]:
nlu_apikey = "rEtSCvVcgGzc0J52bopcij525C2VBT5aOa2L2Qn4Y8-o"
nlu_url = "https://api.eu-de.natural-language-understanding.watson.cloud.ibm.com/instances/fba99f47-a59f-43b1-b9b3-cf2b89e046cf"

### CREDENCIALES API IBM - MARIO
{ \
  "apikey": "rEtSCvVcgGzc0J52bopcij525C2VBT5aOa2L2Qn4Y8-o", \
  "iam_apikey_description": "Auto-generated for key 2c8ee236-7b25-4d41-b0d5-b6f628a882c4", \
  "iam_apikey_name": "Auto-generated service credentials", \
  "iam_role_crn": "crn:v1:bluemix:public:iam::::serviceRole:Manager", \
  "iam_serviceid_crn": "crn:v1:bluemix:public:iam-identity::a/c1eefd8c53d94ef180c679051a5b025c::serviceid:ServiceId-b6d2b06c-54cf-4fb1-9767-2113be5f6240", \
  "url": "https://api.eu-de.natural-language-understanding.watson.cloud.ibm.com/instances/fba99f47-a59f-43b1-b9b3-cf2b89e046cf" \
}

In [189]:
import json
from ibm_watson import NaturalLanguageUnderstandingV1
from ibm_cloud_sdk_core.authenticators import IAMAuthenticator
from ibm_watson.natural_language_understanding_v1 import Features, CategoriesOptions, ConceptsOptions, EmotionOptions, EntitiesOptions, KeywordsOptions, RelationsOptions, SentimentOptions

authenticator = IAMAuthenticator(nlu_apikey)
natural_language_undestanding = NaturalLanguageUnderstandingV1(
    version = '2020-08-01', 
    authenticator = authenticator
)
natural_language_undestanding.set_service_url(nlu_url)

## Importar dataset

In [177]:
import pandas as pd
dataset = pd.read_csv('full-data.csv')

In [178]:
dataset

Unnamed: 0  Unnamed: 0.1  \
0              0             0   
1              1             1   
2              2             2   
3              3             3   
4              4             4   
...          ...           ...   
7804        7804          7804   
7805        7805          7805   
7806        7806          7806   
7807        7807          7807   
7808        7808          7808   

                                                   text  created_at  \
0     @Uber_Support hi guys why all my trips this we...  Wed Aug 11   
1     So are we getting Uber eats league 1 or not @S...  Wed Aug 11   
2     @footballdaily 2-3 years of Messi, a bit of ca...  Wed Aug 11   
3     @Prashan32439454 @Uber_Support @rameshjoshi80 ...  Wed Aug 11   
4     Uber's Gopuff partnership, Drizly deal being p...  Wed Aug 11   
...                                                 ...         ...   
7804  Over 3,800 rides in over 3.5 years one blown t...  Mon Aug 09   
7805  @CoolABADev obviously i will want Messi to suc...  Mon Aug 09   
7806  @catekitchen Shit, that’s absolutely a crap si...  Mon Aug 09   
7807  Simply don’t take a Uber then dumb ass https:/...  Mon Aug 09   
7808  Uber, Lyft prices keep climbing to new highs, ...  Mon Aug 09   

     created_at_time  created_at_hour retweeted  retweet_count  \
0           09:43:42                9        Si              1   
1           09:43:39                9        No              0   
2           09:43:36                9        No              0   
3           09:43:36                9        No              0   
4           09:43:33                9        No              0   
...              ...              ...       ...            ...   
7804        13:01:33               13        No              0   
7805        13:01:27               13        No              0   
7806        13:01:24               13        No              0   
7807        13:00:56               13        No              0   
7808        13:00:38               13        No              0   

      favorite_count  user_ verified  ...  hashtags_text hastags_indices  \
0                  0             NaN  ...             []              []   
1                  0             NaN  ...             []              []   
2                  0             NaN  ...             []              []   
3                  0             NaN  ...             []              []   
4                  0             NaN  ...             []              []   
...              ...             ...  ...            ...             ...   
7804               0             NaN  ...             []              []   
7805               1             NaN  ...             []              []   
7806               2             NaN  ...             []              []   
7807               0             NaN  ...             []              []   
7808               4             NaN  ...             []              []   

     hastags_in_tweet  possitivity_textblob  possitivity_vader  \
0                   0              0.000000             0.0000   
1                   0              0.000000             0.0000   
2                   0             -0.333333             0.5859   
3                   0              0.455000             0.4754   
4                   0              0.000000             0.0000   
...               ...                   ...                ...   
7804                0              0.000000             0.4898   
7805                0              0.114286            -0.0333   
7806                0             -0.250000            -0.7086   
7807                0             -0.187500            -0.8096   
7808                0              0.136364             0.0000   

      possitivity_ibm sentiment_mean  sentiment_norm     sentiment  \
0           -0.912398      -0.304133        0.347934      Negativo   
1            0.000000       0.000000        0.500000        Neutro   
2           -0.807459      -0.18496

In [179]:
dataset.drop(['Unnamed: 0', 'Unnamed: 0.1'], axis = 1, inplace=True)

In [ ]:
dataset.head()

text  created_at  \
0  @Uber_Support hi guys why all my trips this we...  Wed Aug 11   
1  So are we getting Uber eats league 1 or not @S...  Wed Aug 11   
2  @footballdaily 2-3 years of Messi, a bit of ca...  Wed Aug 11   
3  @Prashan32439454 @Uber_Support @rameshjoshi80 ...  Wed Aug 11   
4  Uber's Gopuff partnership, Drizly deal being p...  Wed Aug 11   

  created_at_time  created_at_hour retweeted  retweet_count  favorite_count  \
0        09:43:42                9        Si              1               0   
1        09:43:39                9        No              0               0   
2        09:43:36                9        No              0               0   
3        09:43:36                9        No              0               0   
4        09:43:33                9        No              0               0   

   user_ verified              user_id        user_name  ... status_count  \
0             NaN  1137383922843771008        ShieldVoC  ...       695888   
1             NaN            528288653          __Soini  ...        58398   
2             NaN            826917817          ASHJY97  ...         5974   
3             NaN  1382729072837479936  GurmeetSinghSh6  ...          169   
4             NaN           3646985662           sbwcws  ...       113083   

   hashtags_text  hastags_indices  hastags_in_tweet possitivity_textblob  \
0             []               []                 0             0.000000   
1             []               []                 0             0.000000   
2             []               []                 0            -0.333333   
3             []               []                 0             0.455000   
4             []               []                 0             0.000000   

   possitivity_vader possitivity_ibm sentiment_mean  sentiment_norm  sentiment  
0             0.0000       -0.912398      -0.304133        0.347934   Negativo  
1             0.0000        0.000000       0.000000        0.500000     Neutro  
2             0.5859       -0.807459      -0.184964        0.407518     Neutro  
3             0.4754        0.863584       0.597995        0.798997   Positivo  
4             0.0000       -0.788377      -0.262792        0.368604   Negativo  

[5 rows x 29 columns]

## Obtención del topic

In [ ]:
def topic_ibm(tweet):
  
  try: 
    aux = natural_language_undestanding.analyze(text=tweet, features=Features(categories=CategoriesOptions())).get_result()['categories']
    return aux
  
  except:
    return 'Not Applicable'

In [ ]:
dataset['categories'] = dataset['text'].apply(lambda x: topic_ibm(x))

In [ ]:
dataset.head()

text  created_at  \
0  @Uber_Support hi guys why all my trips this we...  Wed Aug 11   
1  So are we getting Uber eats league 1 or not @S...  Wed Aug 11   
2  @footballdaily 2-3 years of Messi, a bit of ca...  Wed Aug 11   
3  @Prashan32439454 @Uber_Support @rameshjoshi80 ...  Wed Aug 11   
4  Uber's Gopuff partnership, Drizly deal being p...  Wed Aug 11   

  created_at_time  created_at_hour retweeted  retweet_count  favorite_count  \
0        09:43:42                9        Si              1               0   
1        09:43:39                9        No              0               0   
2        09:43:36                9        No              0               0   
3        09:43:36                9        No              0               0   
4        09:43:33                9        No              0               0   

   user_ verified              user_id        user_name  ... hashtags_text  \
0             NaN  1137383922843771008        ShieldVoC  ...            []   
1             NaN            528288653          __Soini  ...            []   
2             NaN            826917817          ASHJY97  ...            []   
3             NaN  1382729072837479936  GurmeetSinghSh6  ...            []   
4             NaN           3646985662           sbwcws  ...            []   

   hastags_indices  hastags_in_tweet  possitivity_textblob possitivity_vader  \
0               []                 0              0.000000            0.0000   
1               []                 0              0.000000            0.0000   
2               []                 0             -0.333333            0.5859   
3               []                 0              0.455000            0.4754   
4               []                 0              0.000000            0.0000   

   possitivity_ibm sentiment_mean sentiment_norm  sentiment  \
0        -0.912398      -0.304133       0.347934   Negativo   
1         0.000000       0.000000       0.500000     Neutro   
2        -0.807459      -0.184964       0.407518     Neutro   
3         0.863584       0.597995       0.798997   Positivo   
4        -0.788377      -0.262792       0.368604   Negativo   

                                          categories  
0  [{'score': 0.693098, 'label': '/travel/special...  
1  [{'score': 0.59693, 'label': '/sports/baseball...  
2   [{'score': 0.785107, 'label': '/sports/soccer'}]  
3  [{'score': 0.685688, 'label': '/science/mathem...  
4  [{'score': 0.778195, 'label': '/news'}, {'scor...  

[5 rows x 30 columns]

In [ ]:
for i in dataset['categories'][7500]:
    print(i)

{'score': 0.618347, 'label': '/business and industrial/business software'}
{'score': 0.605689, 'label': '/technology and computing/operating systems'}
{'score': 0.58508, 'label': '/technology and computing/enterprise technology'}


### Guardar nuevo DataFrame

In [170]:
dataset.to_csv('full-data.csv')

## Feature Engineering con 'categories'

In [23]:
pip install ast

  Using cached AST-0.0.2.tar.gz (19 kB)
    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/jt/rl_2gqv90fqbrwzlwsq9kqcm0000gn/T/pip-install-8z11b3n9/ast_186c19b6e1ef4529b01d65841fb81674/setup.py'"'"'; __file__='"'"'/private/var/folders/jt/rl_2gqv90fqbrwzlwsq9kqcm0000gn/T/pip-install-8z11b3n9/ast_186c19b6e1ef4529b01d65841fb81674/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/jt/rl_2gqv90fqbrwzlwsq9kqcm0000gn/T/pip-pip-egg-info-vpokp4yv
         cwd: /private/var/folders/jt/rl_2gqv90fqbrwzlwsq9kqcm0000gn/T/pip-install-8z11b3n9/ast_186c19b6e1ef4529b01d65841fb81674/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/jt

[1.23, 2.34]

In [15]:
import pandas as pd
dataset = pd.read_csv('full-data.csv')

La variable 'categories' es una lista pero al guardarse en csv, se convierte a string. Al volverlo importar, ya no podemos trabajar con ello como si fuera una lista. Para solucionarlo hay que usar la función ast.literal_eval(). Aplicarla directamente no funcionaria puesto que en algunos tweets tenemos 'Not Applicable' en lugar de una lista. Por ello se crea la siguiente función:

In [87]:
def string_to_list(string):
    from ast import literal_eval
    try:
        x = literal_eval(string)
    except:
        x = 'Not Applicable'
    return x

In [88]:
dataset['categories'] = dataset['categories'].apply(string_to_list)

In [89]:
for cat in dataset['categories'][:20]:
    print(cat)

[{'score': 0.693098, 'label': '/travel/specialty travel/adventure travel'}, {'score': 0.599703, 'label': '/travel/business travel'}, {'score': 0.579002, 'label': '/automotive and vehicles'}]
[{'score': 0.59693, 'label': '/sports/baseball'}, {'score': 0.591677, 'label': '/food and drink'}, {'score': 0.560498, 'label': '/pets/large animals'}]
[{'score': 0.785107, 'label': '/sports/soccer'}]
[{'score': 0.685688, 'label': '/science/mathematics/arithmetic'}, {'score': 0.602089, 'label': '/art and entertainment/shows and events'}, {'score': 0.574684, 'label': '/style and fashion/body art'}]
[{'score': 0.778195, 'label': '/news'}, {'score': 0.647092, 'label': '/business and industrial/advertising and marketing'}, {'score': 0.609852, 'label': '/business and industrial/advertising and marketing/advertising'}]
[{'score': 0.694885, 'label': '/society/crime/sexual offense/rape'}, {'score': 0.671184, 'label': '/society/crime/personal offense'}, {'score': 0.661246, 'label': '/society/crime/personal 

#### Opcion 1
Nos quedamos con la primera etiqueta de la categoría con más score. Cada tweet tendrá una única etiqueta de topic.\
\
**Ej.** [{'score': 0.693098, 'label': '/travel/specialty travel/adventure travel'}, {'score': 0.599703, 'label': '/travel/business travel'}, {'score': 0.579002, 'label': '/automotive and vehicles'}]\
--> **Nos quedamos con 'travel'**

In [90]:
#Hacemos una copia del dataset original para trabajar con esta opcion
dataset_1 = dataset[:100]


In [91]:
def get_topic_1(categories):
    x = categories[0]['label']
    return x.split('/')[1]

In [92]:
dataset_1['topic']=dataset_1['categories'].apply(lambda x: get_topic_1(x))

<ipython-input-92-ca83377ebd66>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset_1['topic']=dataset_1['categories'].apply(lambda x: get_topic_1(x))


In [93]:
dataset_1[['text', 'categories', 'topic']].head()

text  \
0  @Uber_Support hi guys why all my trips this we...   
1  So are we getting Uber eats league 1 or not @S...   
2  @footballdaily 2-3 years of Messi, a bit of ca...   
3  @Prashan32439454 @Uber_Support @rameshjoshi80 ...   
4  Uber's Gopuff partnership, Drizly deal being p...   

                                          categories    topic  
0  [{'score': 0.693098, 'label': '/travel/special...   travel  
1  [{'score': 0.59693, 'label': '/sports/baseball...   sports  
2   [{'score': 0.785107, 'label': '/sports/soccer'}]   sports  
3  [{'score': 0.685688, 'label': '/science/mathem...  science  
4  [{'score': 0.778195, 'label': '/news'}, {'scor...     news

#### Opcion 2
Nos quedamos con la primera etiqueta de cada categoría. \
\
**Ej.** [{'score': 0.693098, 'label': '/travel/specialty travel/adventure travel'}, {'score': 0.599703, 'label': '/travel/business travel'}, {'score': 0.579002, 'label': '/automotive and vehicles'}]\
--> **Nos quedamos con 'travel', 'travel' y 'automotive and vehicles** (luego eliminamos duplicados)

In [94]:
#Hacemos una copia del dataset original para trabajar con esta opcion
dataset_2 = dataset[:100]

In [95]:
def get_topic_2(categories):
    labels = []
    for category in categories:
        x = category['label'].split('/')
        labels.append(x[1])
    return list(set(labels))

In [96]:
dataset_2['topics']=dataset_2['categories'].apply(lambda x: get_topic_2(x))

<ipython-input-96-f6ef7e5265d9>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset_2['topics']=dataset_2['categories'].apply(lambda x: get_topic_2(x))


In [97]:
dataset_2[['text', 'categories', 'topics']].head()

text  \
0  @Uber_Support hi guys why all my trips this we...   
1  So are we getting Uber eats league 1 or not @S...   
2  @footballdaily 2-3 years of Messi, a bit of ca...   
3  @Prashan32439454 @Uber_Support @rameshjoshi80 ...   
4  Uber's Gopuff partnership, Drizly deal being p...   

                                          categories  \
0  [{'score': 0.693098, 'label': '/travel/special...   
1  [{'score': 0.59693, 'label': '/sports/baseball...   
2   [{'score': 0.785107, 'label': '/sports/soccer'}]   
3  [{'score': 0.685688, 'label': '/science/mathem...   
4  [{'score': 0.778195, 'label': '/news'}, {'scor...   

                                              topics  
0                  [travel, automotive and vehicles]  
1                     [food and drink, sports, pets]  
2                                           [sports]  
3  [science, art and entertainment, style and fas...  
4                    [business and industrial, news]

A continuacion, creamos una variable por cada una de las etiquetas en 'topics', es decir, 'topic_1', 'topic_2' y 'topic_3'

In [107]:
def desagregar_topics(lista_topics, n_topic):
    try:
        topic = lista_topics[n_topic-1]
    except:
        topic = 'NA'
    return topic

In [99]:
dataset_2['topic_1'] = dataset_2['topics'].apply(lambda x: desagregar_topics(x, 1))
dataset_2['topic_2'] = dataset_2['topics'].apply(lambda x: desagregar_topics(x, 2))
dataset_2['topic_3'] = dataset_2['topics'].apply(lambda x: desagregar_topics(x, 3))

<ipython-input-99-63b6f2d20988>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset_2['topic_1'] = dataset_2['topics'].apply(lambda x: desagregar_topics(x, 1))
<ipython-input-99-63b6f2d20988>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset_2['topic_2'] = dataset_2['topics'].apply(lambda x: desagregar_topics(x, 2))
<ipython-input-99-63b6f2d20988>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

In [100]:
dataset_2[['text', 'categories', 'topics', 'topic_1', 'topic_2', 'topic_3']].head()

text  \
0  @Uber_Support hi guys why all my trips this we...   
1  So are we getting Uber eats league 1 or not @S...   
2  @footballdaily 2-3 years of Messi, a bit of ca...   
3  @Prashan32439454 @Uber_Support @rameshjoshi80 ...   
4  Uber's Gopuff partnership, Drizly deal being p...   

                                          categories  \
0  [{'score': 0.693098, 'label': '/travel/special...   
1  [{'score': 0.59693, 'label': '/sports/baseball...   
2   [{'score': 0.785107, 'label': '/sports/soccer'}]   
3  [{'score': 0.685688, 'label': '/science/mathem...   
4  [{'score': 0.778195, 'label': '/news'}, {'scor...   

                                              topics                  topic_1  \
0                  [travel, automotive and vehicles]                   travel   
1                     [food and drink, sports, pets]           food and drink   
2                                           [sports]                   sports   
3  [science, art and entertainment, style and fas...                  science   
4                    [business and industrial, news]  business and industrial   

                   topic_2            topic_3  
0  automotive and vehicles                 NA  
1                   sports               pets  
2                       NA                 NA  
3    art and entertainment  style and fashion  
4                     news                 NA

#### Opcion 3
Nos quedamos con todas las etiquetas de la primera categoría. \
\
**Ej.** [{'score': 0.693098, 'label': '/travel/specialty travel/adventure travel'}, {'score': 0.599703, 'label': '/travel/business travel'}, {'score': 0.579002, 'label': '/automotive and vehicles'}]\
--> **Nos quedamos con 'travel', 'specialty travel', 'adventure travel'**\
En este caso, debemos tener tantas columnas como el número máximo de etiquetas en una fila

In [103]:
dataset_3 = dataset[:100]

In [168]:
dataset_3['categories'][0][0]['label']

'/travel/specialty travel/adventure travel'

In [104]:
def get_topic_3(categories):
    x = categories[0]['label']
    x = x.split('/')
    x.remove('')
    return x

In [105]:
dataset_3['topics']=dataset_3['categories'].apply(lambda x: get_topic_3(x))

<ipython-input-105-c40aaccecc86>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset_3['topics']=dataset_3['categories'].apply(lambda x: get_topic_3(x))


In [108]:
dataset_3['topic_1'] = dataset_3['topics'].apply(lambda x: desagregar_topics(x, 1))
dataset_3['topic_2'] = dataset_3['topics'].apply(lambda x: desagregar_topics(x, 2))
dataset_3['topic_3'] = dataset_3['topics'].apply(lambda x: desagregar_topics(x, 3))

<ipython-input-108-98f2389c8cf2>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset_3['topic_1'] = dataset_3['topics'].apply(lambda x: desagregar_topics(x, 1))
<ipython-input-108-98f2389c8cf2>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dataset_3['topic_2'] = dataset_3['topics'].apply(lambda x: desagregar_topics(x, 2))
<ipython-input-108-98f2389c8cf2>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value inst

In [109]:
dataset_3[['text', 'categories', 'topics', 'topic_1', 'topic_2', 'topic_3']].head()

text  \
0  @Uber_Support hi guys why all my trips this we...   
1  So are we getting Uber eats league 1 or not @S...   
2  @footballdaily 2-3 years of Messi, a bit of ca...   
3  @Prashan32439454 @Uber_Support @rameshjoshi80 ...   
4  Uber's Gopuff partnership, Drizly deal being p...   

                                          categories  \
0  [{'score': 0.693098, 'label': '/travel/special...   
1  [{'score': 0.59693, 'label': '/sports/baseball...   
2   [{'score': 0.785107, 'label': '/sports/soccer'}]   
3  [{'score': 0.685688, 'label': '/science/mathem...   
4  [{'score': 0.778195, 'label': '/news'}, {'scor...   

                                         topics  topic_1           topic_2  \
0  [travel, specialty travel, adventure travel]   travel  specialty travel   
1                            [sports, baseball]   sports          baseball   
2                              [sports, soccer]   sports            soccer   
3            [science, mathematics, arithmetic]  science       mathematics   
4                                        [news]     news                NA   

            topic_3  
0  adventure travel  
1                NA  
2                NA  
3        arithmetic  
4                NA

#### Opcion 4
Nos quedamos con todas las etiquetas de cada categoría. \
\
**Ej.** [{'score': 0.693098, 'label': '/travel/specialty travel/adventure travel'}, {'score': 0.599703, 'label': '/travel/business travel'}, {'score': 0.579002, 'label': '/automotive and vehicles'}]\
--> **Nos quedamos con 'travel', 'specialty travel', 'adventure travel', 'travel', 'business travel' 'automotive and vehicles'** (luego eliminamos duplicados)\
En este caso, debemos tener tantas columnas como el número máximo de etiquetas en una fila

In [88]:
#Hacemos una copia del dataset original para trabajar con esta opcion
dataset_4 = dataset